Backpack-SwiftUI/SponsoredBanner/Classes/BPKSponsoredBanner.swift

```diff
@@ -0,0 +1,171 @@
+    let backgroundColor: Color
+    
+    @State var showTerms: Bool
+    
+    public init(
+        logo: Image? = nil,
+        title: String? = nil,
+        subheadline: String? = nil,
+        callToAction: CallToAction? = nil,
+        bodyText: String? = nil,
+        variant: Variant,
+        backgroundColor: Color,
+        showTerms: Bool = false
+    ) {
+        self.logo = logo
+        self.title = title
+        self.subheadline = subheadline
+        self.callToAction = callToAction
+        self.bodyText = bodyText
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.showTerms = showTerms
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            topView
+                .padding(.base)
+                .background(backgroundColor)
+                .zIndex(1)
+                .onTapGesture { toggleBodyText() }
+            if let bodyText = bodyText, showTerms {
+                BPKText(bodyText, style: .caption)
+                    .lineLimit(4)
+                    .frame(maxWidth: .infinity)
+                    .padding(.base)
+                    .background(Color.clear)
+                    .zIndex(0)
+            }
+        }
+        .background(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+                .foregroundColor(.canvasContrastColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+    }
+    
+    private var topView: some View {
+        HStack(spacing: .md) {
+            HStack {
+                if let logo = logo {
+                    logo
+                        .resizable()
+                        .scaledToFit()
+                        .frame(maxHeight: 22)
+                }
+                titlesView
+                Spacer()
+                if let callToAction = callToAction {
+                    BPKText(callToAction.text, style: .caption)
+                        .foregroundColor(variant.color)
+                }
+            }
+            .accessibilityElement(children: .combine)
+            if callToAction?.showIcon == true {
+                iconView
+            }
+        }
+    }
+    
+    private var titlesView: some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            if let title = title {
+                BPKText(title, style: .label2)
+                    .foregroundColor(variant.color)
+            }
+            if let subheadline = subheadline {
+                BPKText(subheadline, style: .caption)
+                    .foregroundColor(variant.color)
+            }
+        }
+    }
+    
+    private var iconView: some View {
+        ZStack {
+            Text("")
+            BPKIconView(.informationCircle)
+                .foregroundColor(variant.color)
+        }
+        .accessibilityAddTraits(.isButton)
+        .accessibilityValue(callToAction?.accessibilityLabel ?? "")
```

Comment:
Does accessibility value with an empty string make sense? Can't we use `ConditionalModifier` here?

Reply:
Did a similar thing, check latest PR 😄